### DISCUSSION TITLE
Grant a privilege to a role

### DISCUSSION THREAD

**Discussion by arkim822** [milvus-io/milvus#31246] [Q&A and General discussion] [answered]
Hello. I am getting this error back when I try to grant a privilege to the role I've created.

`RPC error: [grant_privilege], <MilvusException: (code=65535, message=fail to get authorization from the md, authorize:[authorization])>, <Time:{'RPC start': '2024-03-13 10:31:05.579118', 'RPC error': '2024-03-13 10:31:05.602571'}>`

I have tried to search the discussions but have not been able to find anything. Below, is the code I'm running.

```
from pymilvus import (
    connections,
    Role,
    utility,
    FieldSchema, CollectionSchema, DataType,
    Collection,
)

connections.connect("default", host="some_host", port="some_port")
collection_name = 'Some_Collection'

utility.create_user('admin', 'password', using="default")

role_name = "Master"
role = Role(role_name)

role.grant("Collection", 'Some_Collection', "Search")
```

**Comment by yhmo** [comment, marked as answer]
I guess your milvus server authentication is not enabled: https://milvus.io/docs/authenticate.md

After authentication is enabled, the connection will require a username/password to login. But in your script, you didn't pass any username/password.

Normal workflow:
1. Set common.security.authorizationEnabled in milvus.yaml as true
2. Start milvus server
3. Log in with root user
```
from pymilvus import connections
connections.connect(
    alias='default',
    host='localhost',
    port='19530',
    user='root',
    password='Milvus')
```
4. Create a role, grant privileges to the role
```
    role = Role(name="XXX")
    if not role.is_exist():
        role.create()
        role.grant(object="Collection", object_name="*", privilege="*")
        role.grant(object="Global", object_name="*", privilege="*")
```

5. Create a new user, add user to role
```
    utility.create_user(user="AAA", password="123456")
    role.add_user(username="AAA")
```

Note: be careful if you are using different database name for role/user authentication, read this doc: https://milvus.io/docs/manage_databases.md#Use-the-RBAC-with-database

**Reply by arkim822** [reply]
Thank you! This has worked 👍
